File: packages/vintent/src/modules/shells.ts
import type { ShellType } from "@/modules/shells/types";
import { HeatmapCorrelationShell } from "@/modules/shells/heatmap-correlation";
import { HistogramShell } from "@/modules/shells/histogram";
import { BoxPlotShell } from "@/modules/shells/boxplot";
import { DensityShell } from "@/modules/shells/density";
import { ScatterShell } from "@/modules/shells/scatter";
import { BarCountShell } from "@/modules/shells/bar-count";
import { BarSeriesShell } from "./shells/bar-series";
import { BarValueShell } from "./shells/bar-value";
import { BarAggregateShell } from "./shells/bar-aggregate";
import { LineTimeShell } from "./shells/line-time";
import { LineGenericShell } from "./shells/line-generic";
import { StackedAreaShell } from "./shells/stacked-area";

export const shells: Record<string, ShellType> = {
    bar_aggregate: new BarAggregateShell(),
    bar_count: new BarCountShell(),
    bar_series: new BarSeriesShell(),
    bar_value: new BarValueShell(),
    boxplot: new BoxPlotShell(),
    density: new DensityShell(),
    heatmap_correlation: new HeatmapCorrelationShell(),
    histogram: new HistogramShell(),
    line_generic: new LineGenericShell(),
    line_time: new LineTimeShell(),
    scatter: new ScatterShell(),
    stacked_area: new StackedAreaShell(),
};

File: packages/vintent/vintent/vintent/core/modules/shells.py
from .templates.bar_aggregate import BarAggregateShell
from .templates.bar_count import BarCountShell
from .templates.bar_series import BarSeriesShell
from .templates.bar_value import BarValueShell
from .templates.box_plot import BoxPlotShell
from .templates.density import DensityShell
from .templates.heatmap_correlation import HeatmapCorrelationShell
from .templates.histogram import HistogramShell
from .templates.line_generic import LineGenericShell
from .templates.line_time import LineTimeShell
from .templates.scatter import ScatterShell
from .templates.stacked_area import StackedAreaShell

shells = {
    "bar_aggregate": BarAggregateShell(),
    "bar_count": BarCountShell(),
    "bar_series": BarSeriesShell(),
    "bar_value": BarValueShell(),
    "box_plot": BoxPlotShell(),
    "density": DensityShell(),
    "heatmap_correlation": HeatmapCorrelationShell(),
    "histogram": HistogramShell(),
    "line_generic": LineGenericShell(),
    "line_time": LineTimeShell(),
    "scatter": ScatterShell(),
    "stacked_area": StackedAreaShell(),
}
